book/surface_flows.md

```diff
@@ -18,44 +18,59 @@ Some operators have more than one input _port_ that can be referenced by `->`. F
 merges the contents of many flows, so it can have an abitrary number of input ports. Some operators have multiple outputs, notably [`tee`](./surface_ops.gen.md#tee),
 which has an arbitrary number of outputs.
 
-In the syntax, we distinguish input ports via an _indexing prefix_ number
-in square brackets before the operator name (e.g. `[0]merge(...)` and `[1]merge(...)`). We
-distinguish output ports by an _indexing suffix_ (e.g. `tee[0]`).
+In the syntax, we optionally distinguish input ports via an _indexing prefix_ number
+in square brackets before the name (e.g. `[0]my_merge` and `[1]my_merge`). We
+distinguish output ports by an _indexing suffix_ (e.g. `my_tee[0]`).
 
 Here is an example that tees one flow into two, handles each separately, and then merges them to print out the contents in both lowercase and uppercase:
 ```rust,ignore
-source = source_iter(vec!["Hello", "world"]) -> tee();
-print = merge() -> for_each(|x| println!("{}", x));
-source[0] -> map(|x: &str| x.to_uppercase()) -> [0]print;
-source[1] -> map(|x: &str| x.to_lowercase()) -> [1]print;
+my_tee = source_iter(vec!["Hello", "world"]) -> tee();
+my_tee[0] -> map(|x: &str| x.to_uppercase()) -> [0]my_merge;
```

Comment:
Can we drop indices for `tee` and `merge` in this example? It's confusing to say port indices are optional but then use the right after. Or at least show an example w/ without them or note in a comment that they are not needed.

Reply:
Yeah